hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java

```diff
@@ -1153,9 +1087,51 @@ public void run() {
             + connections.size());
     }
 
+    /**
+     * A thread to write rpc requests to the socket.
+     */
+    private class RpcRequestSender implements Runnable {
+      @Override
+      public void run() {
+        while (!shouldCloseConnection.get()) {
+          ResponseBuffer buf = null;
+          try {
+            Pair<Call, ResponseBuffer> pair = rpcRequestQueue.take();
+            if (shouldCloseConnection.get()) {
+              return;
+            }
+            buf = pair.getRight();
+            synchronized (ipcStreams.out) {
+              if (LOG.isDebugEnabled()) {
+                Call call = pair.getLeft();
+                LOG.debug(getName() + " sending #" + call.id
+                    + " " + call.rpcRequest);
```

Comment:
Maybe we can use `{}`, such as:
```
LOG.debug("{} sending #{} {}.", getName(), call.id, call.rpcRequest);
```

Reply:
Ok, I'll do that. However that code is just moving around in this patch and in general I prefer to not mix improvements and bug fixes.